content/documentation/staging/tutorial/05-connect/04-tcp-traffic-shifting.adoc

```diff
@@ -0,0 +1,74 @@
+---
+title: TCP Traffic Shifting
+weight: 53
+---
+:icons: font
+:sectlinks:
+
+The Travel Demo application has a database service used by several services deployed in the *travel-agency* namespace.
+
+At some point in the lifecycle of the application, the telemetry shows that the database service degrades and starts to increase the average response time.
+
+This is a common situation and in this case database specialist suggests that data are growing and the original indexes design needs to be updated.
+
+Our database specialist is suggesting two approaches and proposes to prepare two versions of the database service to test which may work better.
+
+This step will show how the "Traffic Shifting" strategy can be applied to TCP services to test which database new indexes strategy works better.
+
+NOTE: *[1]* Deploy *mysqldb-v2* and *mysqldb-v3* workloads
+
+To deploy the new versions of the *mysqldb* service follow the steps:
+
+[source,bash]
+----
+kubectl apply -f <(curl -L https://raw.githubusercontent.com/kiali/demos/master/travels/mysql-v2.yaml) -n travel-agency
+kubectl apply -f <(curl -L https://raw.githubusercontent.com/kiali/demos/master/travels/mysql-v3.yaml) -n travel-agency
+----
+
+NOTE: *[2]* Use the TCP Traffic Shifting Wizard on *mysqldb* service
+
+++++
+<a class="image-popup-fit-height" href="/images/tutorial/05-04-tcp-traffic-shifting-action.png" title="TCP Traffic Shifting Action">
+    <img src="/images/tutorial/05-04-tcp-traffic-shifting-action.png" style="display:block;margin: 0 auto;" />
+</a>
+++++
+
+{nbsp} +
+Create a scenario with 80% of the traffic distributed on *mysqldb-v1* workload and 10% of the traffic distributed on *mysqldb-v2* and *mysqldb-v3*.
+
+++++
+<a class="image-popup-fit-height" href="/images/tutorial/05-04-tcp-split-traffic.png" title="TCP Split Traffic">
+    <img src="/images/tutorial/05-04-tcp-split-traffic.png" style="display:block;margin: 0 auto;" />
+</a>
+++++
+
+NOTE: *[3]* Examine Traffic Shifting distribution from the *travels-agency* Graph
+
+++++
+<a class="image-popup-fit-height" href="/images/tutorial/05-04-tcp-graph.png" title="MysqlDB Graph">
+    <img src="/images/tutorial/05-04-tcp-graph.png" style="display:block;margin: 0 auto;" />
+</a>
+++++
+
+{nbsp} +
+Note that TCP telemetry has different type of metrics, as "Request Distribution" is only available for HTTP/gRPC services, for this service we need to use "Request Rate" to study the distribution of data between *mysqldb* workloads.
+
+NOTE: *[4]* Compare *mysqldb* workload and study new indexes proposed in *mysqldb-v2* and *mysqldb-v3*
+
+TCP services have different telemetry, but it's still grouped by versions allowing to compare and study differences on the patterns of *mysqldb-v2* and *mysqldb-v3*.
```

Comment:
TCP services have different telemetry but it's still grouped by versions, allowing the user to compare and study pattern differences for *mysqldb-v2* and *mysqldb-v3*.